client/src/components/AddTag/AddTagModal/index.tsx

```diff
@@ -0,0 +1,300 @@
+import ModalHeader from '../../../vendor/react-store/components/View/Modal/Header';
+import { RestRequest } from '../../../vendor/react-store/utils/rest';
+
+import NonFieldErrors from '../../../vendor/react-store/components/Input/NonFieldErrors';
+import Faram, {
+    FaramErrors,
+    FaramValues,
+    FaramSchema,
+} from '../../../vendor/react-store/components/Input/Faram';
+import { requiredCondition } from '../../../vendor/react-store/components/Input/Faram/validations';
+import SelectInput from '../../../vendor/react-store/components/Input/SelectInput';
+import TextArea from '../../../vendor/react-store/components/Input/TextArea';
+import TextInput from '../../../vendor/react-store/components/Input/TextInput';
+import LoadingAnimation from '../../../vendor/react-store/components/View/LoadingAnimation';
+
+import {
+    projectsSelector,
+    tagSelector,
+    setTagAction,
+} from '../../../redux';
+import {
+    Project,
+    RootState,
+    Tag,
+} from '../../../redux/interface';
+import { AddTagParams } from '../../../rest/interface';
+
+import { iconNames } from '../../../constants';
+
+import TagPostRequest from '../requests/TagPostRequest';
+import TagGetRequest from '../requests/TagGetRequest';
+import TagPutRequest from '../requests/TagPutRequest';
+
+import * as styles from './styles.scss';
+
+interface OwnProps{
+    onClose(): void;
+    projectId?: number;
+    onTagCreate?(taskId: number): void;
+    disabledProjectChange?: boolean;
+    tagId?: number;
+}
+interface PropsFromState{
+    projects: Project[];
+    tag?: Tag;
+}
+interface PropsFromDispatch {
+    setTag(value: Tag): void;
+}
+
+type Props = OwnProps & PropsFromState & PropsFromDispatch;
+
+interface State {
+    faramErrors: FaramErrors;
+    faramValues: FaramValues;
+    pending: boolean;
+    pristine: boolean;
+}
+
+export class AddTagModal extends React.PureComponent<Props, State> {
+    addTagRequest: RestRequest;
+    editTagRequest: RestRequest;
+    tagRequest: RestRequest;
+    schema: FaramSchema;
+
+    static keySelector = (d: Project) => d.id;
+    static labelSelector = (d: Project) => d.title;
+
+    constructor(props: Props) {
+        super(props);
+
+        this.schema = {
+            fields: {
+                project: [requiredCondition],
+                title: [requiredCondition],
+                description: [],
+            },
+        };
+
+        let faramValues = props.tag ? props.tag : {};
+        if (props.projectId !== undefined) {
+            faramValues = {
+                project: props.projectId,
+            };
+        }
+        this.state = {
+            faramValues,
+            faramErrors: {},
+            pending: true,
+            pristine: true,
+        };
+    }
+
+    componentWillMount() {
+        const { tagId } = this.props;
+        if (tagId) {
+            this.startRequestForTag(tagId);
+        } else {
+            this.setState({ pending: false });
```

Comment:
pending should be set from the constructor
Use componentDidMount instead of componentWillMount